File: roast/src/renderer/scene.rs
use crate::renderer::camera::Camera;
use crate::renderer::MeshId;

/// See `ai.arcblroth.boss.render.Scene`
/// A Scene stores all the data passed from the
/// high-level code needed to render a single
/// frame of Bosstrove's Revenge.
pub struct Scene {
    pub camera: Camera,
    pub scene_meshes: Vec<MeshId>,
    pub gui_meshes: Vec<MeshId>,
}

File: core/src/main/kotlin/ai/arcblroth/boss/render/Scene.kt
package ai.arcblroth.boss.render

/**
 * A `Scene` stores all the information needed to render
 * a frame of Bosstrove's Revenge.
 */
class Scene(
    /**
     * The camera to use for rendering the [sceneMeshes].
     */
    val camera: Camera = Camera(),

    /**
     * The main array of meshes to render. The backend
     * will render these in a three-dimensional context
     * with depth and lighting.
     */
    val sceneMeshes: ArrayList<Mesh> = arrayListOf(),

    /**
     * An array of meshes to render on top of the GUI.
     * The backend will render these in a two-dimensional
     * context, with later meshes rendering on top of
     * earlier ones.
     */
    val guiMeshes: ArrayList<Mesh> = arrayListOf(),
)
